ocis-pkg/store/options.go

```diff
@@ -1,50 +1,59 @@
 package store
 
-import "time"
-
-// Option provides an option to configure the store
-type Option func(*Options)
-
-// Type defines the type of the store
-func Type(typ string) Option {
-	return func(o *Options) {
-		o.Type = typ
+import (
+	"context"
+	"time"
+
+	"go-micro.dev/v4/store"
+)
+
+type typeContextKey struct{}
+
+// Type determines the implementation:
+//   - "memory", for a in-memory implementation, which is also the default if noone matches
+//   - "noop", for a noop store (it does nothing)
+//   - "etcd", for etcd
+//   - "nats-js" for nats-js, needs to have TTL configured at creation
+//   - "redis", for redis
+//   - "redis-sentinel", for redis-sentinel
+//   - "ocmem", custom in-memory implementation, with fixed size and optimized prefix
+//     and suffix search
+func Type(val string) store.Option {
+	return func(o *store.Options) {
+		if o.Context == nil {
+			o.Context = context.Background()
+		}
+
+		o.Context = context.WithValue(o.Context, typeContextKey{}, val)
```

Comment:
Confused: Why is this all done via the `context`? Using a `ctx` to pass configuration values is not recommended. Can't we have custom options?

Reply:
Can we expand the micro Store interface for Options so that we can enrich it with custom options that our intermediate store.Create() can interpret? The micro Store implementations use the Context to pass driver specific config options. 

I wanted to go with that pattern instead of wrapping and unwrapping micro Options with ocis Options, which seemed far more confusing.